elixir/apps/domain/lib/domain/resources.ex

```diff
@@ -46,6 +47,26 @@ defmodule Domain.Resources do
     end
   end
 
+  def count_resources_for_gateway(%Gateways.Gateway{} = gateway, %Auth.Subject{} = subject) do
+    count =
+      Resource.Query.all()
+      |> Resource.Query.by_account_id(subject.account.id)
```

Comment:
We don't need to filter by `account_id` explicitly, just reuse authorization logic that is used in other functions:
```suggestion
      |> Authorizer.for_subject(subject)
```